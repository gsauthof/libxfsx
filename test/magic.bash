#!/bin/bash

set -eux

: ${awk:=awk}
: ${file:=file}
: ${diff:=diff}
: ${xargs:=xargs}
: ${default_magic:=/usr/share/misc/magic}
: ${out_dir:=out}
src_dir="$1"

out="$out_dir"/magic.out
ref="$src_dir"/test/ref/magic.out
list="$src_dir"/test/ref/magic.list
magic="$src_dir"/config/ber_magic


"$awk" '{printf("'"$src_dir"'/%s\n", $0);}' "$list" | "$xargs" -d '\n' "$file" -m "$default_magic":"$magic" -bz > "$out"


"$diff" -u "$ref" "$out"


