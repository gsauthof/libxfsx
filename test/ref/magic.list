test/in/nrt_2_1.ber
test/in/tap_3_12_valid.ber
test/in/rap_ack.ber
test/in/asn1c/examples/sample.source.TAP3/sample-DataInterChange-1.ber
test/in/rap_mini.ber
test/in/tap_notification.ber
