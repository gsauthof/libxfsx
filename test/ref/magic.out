NRT 2.1 (TD.35, Near Real Time Roaming Data Exchange)
TAP 3.12 Batch (TD.57, Transferred Account)
RAP Acknowledgement (TD.32, Returned Account Procedure)
TAP 3.10 Batch (TD.57, Transferred Account)
RAP 1.5 Batch (TD.32, Returned Account Procedure), TAP 3.12
TAP 3.9 Notification (TD.57, Transferred Account)
